src/registry/select/edit.tsx

```diff
@@ -0,0 +1,206 @@
+import {SelectComponentSchema} from '@open-formulieren/types';
+import {Option} from '@open-formulieren/types/lib/formio/common';
+import {useFormikContext} from 'formik';
+import isEqual from 'lodash.isequal';
+import {useLayoutEffect} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {
+  BuilderTabs,
+  ClearOnHide,
+  Description,
+  Hidden,
+  IsSensitiveData,
+  Key,
+  Label,
+  Multiple,
+  PresentationConfig,
+  Registration,
+  SimpleConditional,
+  Tooltip,
+  Translations,
+  Validate,
+  ValuesConfig,
+  ValuesTranslations,
+  useDeriveComponentKey,
+} from '@/components/builder';
+import {LABELS} from '@/components/builder/messages';
+import {Select, TabList, TabPanel, Tabs} from '@/components/formio';
+import {getErrorNames} from '@/utils/errors';
+
+import {EditFormDefinition} from '../types';
+import {checkIsManualOptions} from './helpers';
+
+/**
+ * Form to configure a Formio 'select' type component.
+ */
+const EditForm: EditFormDefinition<SelectComponentSchema> = () => {
+  const intl = useIntl();
+  const [isKeyManuallySetRef, generatedKey] = useDeriveComponentKey();
+  const {values, errors, setFieldValue} = useFormikContext<SelectComponentSchema>();
+  const {
+    openForms: {dataSrc},
+    defaultValue,
+  } = values;
+
+  const erroredFields = Object.keys(errors).length
+    ? getErrorNames<SelectComponentSchema>(errors)
+    : [];
+  // TODO: pattern match instead of just string inclusion?
+  // TODO: move into more generically usuable utility when we implement other component
+  // types
+  const hasAnyError = (...fieldNames: string[]): boolean => {
+    if (!erroredFields.length) return false;
+    return fieldNames.some(name => erroredFields.includes(name));
+  };
+
+  Validate.useManageValidatorsTranslations<SelectComponentSchema>(['required']);
+
+  const isManualOptions = checkIsManualOptions(values);
+  const options = isManualOptions ? values.data?.values || [] : [];
+
+  // Ensure that form state is reset if the values source changes.
+  useLayoutEffect(() => {
+    if (dataSrc !== 'variable' || isEqual(defaultValue, {})) return;
+    setFieldValue('defaultValue', '');
+  }, [dataSrc]);
```

Comment:
Should we watch for `defaultValue` as well?

Reply:
I don't think so, because the reset behaviour should only kick in when the `dataSrc` value changes.

If the `dataSrc` is set to `manual` and the user changes the default value in the dropdown, this effect would cause it to be reset to no default value which entirely beats the point of having a default value field :P

There is a bug here however - the comparison should be against `''` instead of `{}` in the guard clause.

The idea is that whenever the user:

* switch from variable -> manual: nothing to do, since it's not possible to set a default value for 'variable' dataSrc
* switch from manual -> variable: if a default value was set, clear it, since we don't know which values will exist (they come from `itemsExpression`). if no default value was set -> nothing to do/reset

Reply:
This made me realize that the `multiple` behaviour was also not incorporated - which is not relevant for selectboxes or radio, so I've included that in my latest commit.

Reply:
But what about ESLint in this case? As you said it requires every variable used to be watched?

Reply:
yeah, this would be one case where you silence the ESLint warning - complying with the ESLint rule would lead to incorrect behaviour in this case.

It's not a perfect linter rule, unfortunately. The goal of React was also to figure out the dependencies at compile time, but they can't do so reliably (yet) because of usages like this.

We have another case like this in the SDK, where progress indicator on mobile should collapse on navigate: https://github.com/open-formulieren/open-forms-sdk/blob/8117bacd777d785b197703cc9948bce9c9e004fc/src/components/ProgressIndicator/index.js#L30

Letting the hook run when `expanded` changes there basically prevents you from ever opening the progress indicator. Sometimes a certain state should deliberately be excluded from the hook triggers.